src/libarena/lib.rs

```diff
@@ -189,6 +257,7 @@ impl<T> TypedArena<T> {
             if let Some(last_chunk) = chunks.last_mut() {
                 let used_bytes = self.ptr.get() as usize - last_chunk.start() as usize;
                 let currently_used_cap = used_bytes / mem::size_of::<T>();
+                last_chunk.entries = currently_used_cap;
```

Comment:
Looks like this was an existing bug, right? Good find!

Reply:
There previously wasn't a way to end up with unused memory at the end of a chunk for types with destructors, so not an existing bug.

Reply:
No? It looks like a similar case can also happen already in `alloc_slice`:
https://github.com/rust-lang/rust/blob/f694222887cf31f51e68927716c25736e62f037f/src/libarena/lib.rs#L170

Reply:
That requires `Copy` types, so it can't trigger it.